samples/net/cloud/tagoio_http_post/src/main.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tagoio_http_post, CONFIG_TAGOIO_HTTP_POST_LOG_LEVEL);
+
+#include <zephyr.h>
+#include <net/socket.h>
+#include <net/http_client.h>
+#include <random/rand32.h>
+#include <stdio.h>
+
+#include "wifi.h"
+#include "sockets.h"
+
+static struct tagoio_context_t ctx;
+
+static void response_cb(struct http_response *rsp,
+			enum http_final_call final_data,
+			void *user_data)
+{
+	if (final_data == HTTP_DATA_MORE) {
+		LOG_DBG("Partial data received (%zd bytes)", rsp->data_len);
+	} else if (final_data == HTTP_DATA_FINAL) {
+		LOG_DBG("All the data received (%zd bytes)", rsp->data_len);
+	}
+
+	LOG_DBG("Response status %s", rsp->http_status);
+}
+
+static int collect_data(void)
+{
+	#define lower  20000
+	#define upper 100000
+	#define base    1000.00f
+
+	float temp;
+	int size;
+
+	/* Generate a temperature between 20 and 100 celsius degree */
+	temp = ((sys_rand32_get() % (upper - lower + 1)) + lower);
+	ctx.raw_temperature = temp / base;
```

Comment:
Couldn't `raw_temperature` be a local variable here? It isn't used outside of this function, I don't think it needs to be a part of the ` struct tagoio_context`.

Reply:
Yes, it could! Adjusted.